src/view.rs

```diff
@@ -507,18 +507,18 @@ pub fn format_quantity(
 }
 
 fn format_duration(duration: Duration, number_style: impl Display, unit_style: impl Display) -> String {
-    let mut seconds = duration.as_secs();
+    let mut millis = duration.as_millis();
 
-    if seconds == 0 {
+    if millis == 0 {
         return format!("{}0{}s", number_style, unit_style);
     }
 
     let mut string = String::new();
 
-    for (unit, factor) in &[("h", 3600), ("m", 60), ("s", 1)] {
-        if seconds >= *factor {
-            string.push_str(&format!("{}{}{}{}", number_style, seconds / factor, unit_style, unit));
-            seconds %= factor;
+    for (unit, factor) in &[("h", 3600_000), ("m", 60_000), ("s", 1_000), ("ms", 1)] {
+        if millis >= *factor {
+            string.push_str(&format!("{}{}{}{}", number_style, millis / factor, unit_style, unit));
+            millis %= factor;
```

Comment:
I don't think formatting sub-second durations as "100 ms" makes for a good UX. It's probably better to restrict intervals to multiples of 100 ms and then format as "0.2 s" and so on.